src/util/ReactUtils.ts

```diff
@@ -110,27 +110,28 @@ export const findAllByType = (
     types = [getDisplayName(type)];
   }
 
-  toArray(children).forEach((child: React.DetailedReactHTMLElement<any, HTMLElement>) => {
+  toArray(children).forEach(child => {
     const childType = _.get(child, 'type.displayName') || _.get(child, 'type.name');
     if (types.indexOf(childType) !== -1) {
-      result.push(child);
+      result.push(child as DetailedElement);
     }
   });
 
   return result;
-};
+}
+
 /*
  * Return the first matched child by type, return null otherwise.
  * `type` can be a React element class or string.
  */
-export const findChildByType = (
-  children: ReactNode[],
-  type: string,
-): React.DetailedReactHTMLElement<any, HTMLElement> => {
+export function findChildByType<
+  ComponentType extends React.ComponentType,
+  DetailedElement = React.DetailedReactHTMLElement<React.ComponentProps<ComponentType>, HTMLElement>,
+>(children: ReactNode[], type: ComponentType | ComponentType[]): DetailedElement {
   const result = findAllByType(children, type);
 
-  return result && result[0];
-};
+  return (result && result[0]) as never;
```

Comment:
I would rather avoid the cast to `never` here if we could

Instead of creating `DetailedElement` can we just make the return type
`React.DetailedReactHTMLElement<React.ComponentProps<ComponentType>, HTMLElement>`

so the the cast to never can be removed?

Or if we don't want the return type to be so long to do something like
```ts
type DetailedElement<ComponentType extends React.ComponentType> = React.DetailedReactHTMLElement<
  React.ComponentProps<ComponentType>,
  HTMLElement
>;
...
export function findChildByType<ComponentType extends React.ComponentType>(
  children: ReactNode[],
  type: ComponentType | ComponentType[],
): DetailedElement<ComponentType> {
```

Reply:
I removed the return type alongside the `never` casting.
TS infers all correct types from `findAllByType`.